### PR TITLE
Implement EZP-24538: Add language priority handling to Legacy Search

### PR DESCRIPTION
**Pull request by pspanja**
#### Depends on https://github.com/ezsystems/ezpublish-kernel/pull/1428

With this Legacy Search Engine now handles `$languageSettings` parameters as a list of prioritized languages. SQL logic handling this is reimplemented from [Legacy Stack](https://github.com/ezsystems/ezpublish-legacy/blob/master/kernel/classes/ezcontentlanguage.php#L690). Bit shifting part does not use DBAL as it is not supported there (not in Doctrine nor in our Zeta to Doctrine bridge).

Integration tests for this feature are part of standard integration test suite and are therefore executed on all supported platforms (MySQL, PostgreSQL and SQLite).

This implementation also led to complete removal of language code from `Field` sort clause.
Field filters test case previously used by Legacy and Elasticsearch engines is here removed, as this caused some failures with Elasticsearch. Ideally Elasticsearch should still be maintained for features not pertaining to the language settings.

### CODE REVIEW

**Comment by pspanja** [comment]
Review ping @andrerom @bdunogier @glye @lolautruche @yannickroger

File: eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/SortClauseHandler/Field.php [hunk 1]
```diff
@@ -191,8 +174,95 @@ public function applyJoin(SelectQuery $query, SortClause $sortClause, $number)
...
+    /**
+     * Returns a field language join condition for the given $languageSettings.
+     *
+     * @param \eZ\Publish\Core\Persistence\Database\SelectQuery $query
+     * @param array $languageSettings
+     * @param string $fieldTableName
+     *
+     * @return string
+     */
+    protected function getFieldCondition(SelectQuery $query, array $languageSettings, $fieldTableName)
```
Comment (andrerom) [inline eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/SortClauseHandler/Field.php]: this seems to filter on ezcontentobject, so a bit strange to apply this in Field SortClause.

Comment (andrerom) [inline eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/SortClauseHandler/Field.php]: shouldn't this filter be on the main join of ezcontentobject_attribute table? Tought we would avoid joining several times now, or?

Comment (pspanja) [inline eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/SortClauseHandler/Field.php]: TBH I started with that approach, but it was hard to debug while I didn't know if the re-implementation of the filter is correct, so I changed to this one as it seemed easier. Now that https://github.com/ezsystems/ezpublish-kernel/pull/1428 is fixed and we have some tests in place, I can give it another try.

Comment (pspanja) [inline eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/SortClauseHandler/Field.php]: BTW, as far as I can see that implementation will need to use [CASE operator](https://dev.mysql.com/doc/refman/5.0/en/control-flow-functions.html#operator_case) to select sort columns, which is not abstracted and is in fact vendor-specific, but it should be possible.

**Comment by pspanja** [comment]
Closing in favor of https://github.com/ezsystems/ezpublish-kernel/pull/1433, reopened against master.

**Pull request closed without merging**